codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -180,6 +181,27 @@ DefaultHttp2FrameStream newStream() {
         return new DefaultHttp2FrameStream();
     }
 
+    /**
+     * Creates a new outbound/local stream.
+     */
+    DefaultHttp2FrameStream newStream(int streamId) throws Http2Exception {
+        assert ctx.executor().inEventLoop();
+        DefaultHttp2FrameStream stream = new DefaultHttp2FrameStream();
+        frameStreamToInitializeMap.put(streamId, stream);
```

Comment:
This may never get removed if no headers frame is sent for created stream

Reply:
@mostroverkhov it will be removed if `createStream(...)` not throw in this line:

https://github.com/netty/netty/blob/8036204be62d6e1168269a5186d1852c634570f1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java#L501

I guess I should add a comment to make this more clear